frontend/apps/reader/modules/readerrolling.lua

```diff
@@ -884,6 +884,9 @@ function ReaderRolling:updatePos()
         -- document closed since we were scheduleIn'ed
         return
     end
+
+    Device:setIgnoreInput(true) -- avoid ANRs
+
     -- Check if the document has been re-rendered
     local new_rendering_hash = self.ui.document:getDocumentRenderingHash()
```

Comment:
Isn't this a bit too late ? "has been re-rendered", so the long rendering time might have already happened and is done.
I'm not super sure of this, it's possible the re-rendering time "will happen" at paint time, I'm not sure (but it feels like to get this hash, crengine has to have computed the new styles).
Might want to put some logs here and there to check if rendering has not yet started here.

A quick test with
```lua
        logger.warn("aaa")
    -- Check if the document has been re-rendered
    local new_rendering_hash = self.ui.document:getDocumentRenderingHash()
        logger.warn("bbb")
```
shows this:
12/02/21-10:18:07 DEBUG CreDocument: set interline space 95
12/02/21-10:18:09 DEBUG refresh on physical rectangle 0 2 200 7
12/02/21-10:18:10 DEBUG refresh on physical rectangle 0 2 200 7
12/02/21-10:18:11 DEBUG refresh on physical rectangle 0 2 200 7
12/02/21-10:18:12 DEBUG refresh on physical rectangle 0 2 200 7
12/02/21-10:18:12 DEBUG refresh on physical rectangle 0 2 200 7
12/02/21-10:18:13 DEBUG refresh on physical rectangle 0 2 200 7
12/02/21-10:18:13 DEBUG refresh on physical rectangle 0 2 200 7
12/02/21-10:18:14 DEBUG refresh on physical rectangle 0 2 200 7
12/02/21-10:18:14 DEBUG refresh on physical rectangle 0 2 200 7
12/02/21-10:18:15 DEBUG refresh on physical rectangle 0 2 200 7
12/02/21-10:18:16 DEBUG refresh on physical rectangle 0 2 200 7
CRE WARNING: table rows/thead/tfoot re-ordered
12/02/21-10:18:16 DEBUG refresh on physical rectangle 0 2 200 7
12/02/21-10:18:17 DEBUG refresh on physical rectangle 0 2 200 7
12/02/21-10:18:17 WARN  aaa
12/02/21-10:18:17 WARN  bbb
12/02/21-10:18:17 DEBUG rendering hash changed: 24709753 > 3351176581